### ISSUE TITLE
TreeSelect nodes should expand when clicking the parent node

### ISSUE THREAD

**Issue by AshutoshSajan** [ant-design/ant-design#27507]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
to expand the three-nodes we have to specifically click on an arrow icon at the beginning of a tree which causes a bad user experence, instaed of that we can use a node to expand the tree nodes.

### What does the proposed API look like?
we can add an expandable prop  on the tree node if it's true then it should expand the child nodes of the parent treenode, if selctable and disable is true then also it should work

**Comment by afc163** [comment]
Try https://ant.design/components/tree/#components-tree-demo-directory

**Comment by AshutoshSajan** [comment]
the DirectoryTree is not matching the requirement

**Comment by huseyingoztok** [comment]
@yesmeck @zombieJ @yiminghe  please help!

I have a hierarchical data from a university, faculty, department and program. 

I would like to add fields in TreeSelect to facilitate selection, such as All Faculties, All Departments and All Programs. However, it is not possible to do this because TreeSelectte nodes are interdependent. Even if I lost the connection with treeCheckStrictly, I couldn't do it. What kind of organization should I provide in the nodes? 

In summary my request; Parent's selection of only children who belong to him. Children do not choose their children.

I will be grateful if you could help me. Have a nice day.

[Example](https://res.cloudinary.com/do2ebdrez/image/upload/v1604481024/treeselect_1.png)

**Comment by arpitmalik** [comment]
Any Update on this?

**Comment by zombieJ** [comment]
As @afc163  mentioned. You can copy the logic of `title` expand. We may not extends `expandable` props currently.

ref: https://github.com/ant-design/ant-design/blob/215d058fe55c7539f0cc3d016bcff04670e09ed2/components/tree/DirectoryTree.tsx#L127

**Comment by coldice945** [comment]
This can be achieved by modifying the styles.

index.js

```jsx
import { TreeSelect } from 'antd';
import classNames from 'classnames';
import './index.less';

<TreeSelect
  {...restProps}
  dropdownClassName={classNames('custom-select-dropdown', dropdownClassName)}
/>
```

index.less

```less
.custom-select-dropdown {
  .@{ant-prefix}-select-tree {
    .@{ant-prefix}-select-tree-treenode {

      .@{ant-prefix}-select-tree-switcher {
        position: absolute;
        width: 100%;
        text-align: left;
        z-index: 2;
        left: 7px;

        .@{ant-prefix}-select-tree-switcher-icon {
          visibility: hidden;
        }

        &:hover+.@{ant-prefix}-select-tree-node-content-wrapper {
          background-color: @item-hover-bg;
        }
      }

      .@{ant-prefix}-select-tree-node-content-wrapper {
        margin-left: 24px;
      }

      &.@{ant-prefix}-select-tree-treenode-switcher-open {
        .@{ant-prefix}-select-tree-node-content-wrapper {

          &:before {
            content: '';
            position: absolute;
            top: 10px;
            left: -16px;
            width: 0;
            height: 0;
            border-left: 4px solid transparent;
            border-right: 4px solid transparent;
            border-top: 4px solid #161616;
            z-index: 1;
            transform: rotate(0deg);
          }
        }
      }

      &.@{ant-prefix}-select-tree-treenode-switcher-close {
        .@{ant-prefix}-select-tree-node-content-wrapper {
          &:before {
            content: '';
            position: absolute;
            top: 10px;
            left: -16px;
            width: 0;
            height: 0;
            border-left: 4px solid transparent;
            border-right: 4px solid transparent;
            border-top: 4px solid #161616;
            z-index: 1;
            transform: rotate(-90deg);
          }
        }
      }
    }
  }
}
```
